File: app/src/main/java/ua/hospes/rtm/core/ui/AbsFragment.java
package ua.hospes.rtm.core.ui;

import android.app.Activity;
import android.content.Context;

import androidx.annotation.StringRes;
import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;

public abstract class AbsFragment extends Fragment {
    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        updateABTitle(getActivity(), setActionBarTitle());
    }

    @StringRes
    protected abstract int setActionBarTitle();

    protected void updateABTitle(Activity activity, @StringRes int title) {
        if (title == -1 || title == 0 || !(activity instanceof AppCompatActivity)) return;

        ActionBar ab = ((AppCompatActivity) activity).getSupportActionBar();
        if (ab != null) ab.setTitle(getString(title));
    }
}

File: app/src/main/java/ua/hospes/rtm/core/ui/AbsFragment.kt
package ua.hospes.rtm.core.ui

import android.app.Activity
import android.content.Context
import androidx.annotation.StringRes
import androidx.appcompat.app.AppCompatActivity
import ua.hospes.rtm.core.DiFragment

abstract class AbsFragment : DiFragment {

    constructor() : super()
    constructor(contentLayoutId: Int) : super(contentLayoutId)


    override fun onAttach(context: Context) {
        super.onAttach(context)
        updateABTitle(activity, setActionBarTitle())
    }

    @StringRes
    protected abstract fun setActionBarTitle(): Int

    protected fun updateABTitle(activity: Activity?, @StringRes title: Int) {
        if (title == -1 || title == 0 || activity !is AppCompatActivity) return

        val ab = activity.supportActionBar
        if (ab != null) ab.title = getString(title)
    }
}